library/src/main/java/org/codice/security/saml/EntityInformation.java

```diff
@@ -94,45 +90,38 @@ Binding getBinding(AuthnRequest request, Binding preferred) {
     return preferred != null ? preferred : PREFERRED_BINDING;
   }
 
-  public Map<Binding, String> getAssertionConsumerServicesURLs() {
-    return assertionConsumerServices
-        .entrySet()
-        .stream()
-        .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey(), entry.getKey().getUri()))
-        .collect(
-            Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
-  }
-
   public ServiceInfo getAssertionConsumerService(Binding binding) {
     return assertionConsumerServices.get(binding);
   }
 
   public ServiceInfo getAssertionConsumerService(
       AuthnRequest request, Binding preferred, Integer index) {
     ServiceInfo si = null;
+
     if (request != null && request.getAssertionConsumerServiceURL() != null) {
-      return getAssertionConsumerServiceInfoByUrl(request);
+      si = getAssertionConsumerServiceInfoByUrl(request);
```

Comment:
❓ Is there any reason for the ordering of `ByUrl`, `ByIndex`, and `ByProtocolBinding` when trying to obtain the `acsUrl`?

Reply:
Unimportant.